tests/system/common.py

```diff
@@ -83,3 +85,43 @@ def wait_for_mesos_endpoint(timeout_sec=timedelta(minutes=5).total_seconds()):
     it returns false"""
 
     return shakedown.time_wait(lambda: shakedown.mesos_available_predicate(), timeout_seconds=timeout_sec)
+
+
+def metronome_available_predicate():
+    url = metronome_api_url()
+    try:
+        response = http.get(url)
+        return response.status_code == 200
+    except Exception as e:
+        return False
+
+
+def metronome_api_url():
+    return shakedown.dcos_url_path("/service/metronome/v1/jobs")
+
+
+def assert_job_run(client, job_id, runs_number=1, tasks_number=1):
+    job_runs_count = len(client.get_runs(job_id))
+    assert job_runs_count == runs_number, "Expecting 1 job run but found {} for job {}.".format(job_runs_count, job_id)
+    job_run_tasks = client.get_runs(job_id)[0]["tasks"]
+    assert len(job_run_tasks) == tasks_number, "Expecting 1 job run task but found {} for job {}: {}.".format(len(job_run_tasks), job_id, job_run_tasks)
+
+
+def job_run_predicate(job_id, run_id):
+    run = metronome.create_client().get_run(job_id, run_id)
+    return run["status"] == "ACTIVE" or run["status"] == "SUCCESS"
+
+
+def wait_for_job_started(job_id, run_id, timeout=120):
+    "Verifies that a job with given run_id is in state running or finished. "
+    shakedown.time_wait(lambda: job_run_predicate(job_id, run_id),
+              timeout)
+
+
+def assert_wait_for_no_additional_tasks(client, job_id, timeout=20, tasks_count=1):
+    """ Starting Metronome and all its actors takes some time and there is no way how to query API to figure out it finished.
+        Here we wait for given time and then assert that expected tasks count matches.
+        This covers a bug regression of METRONOME-100
+    """
+    time.sleep(timeout)
```

Comment:
I did not figure out a way hot to get rid of this sleep part.

I need to wait for some actors to be spawned up after restart and I am not sure there is a way hot to figure out from API that it already happened...

Reply:
So we wait to see if metronome decides to run the job? I guess we could retry for a while and fail if we find a new `scheduled`, `active` or `history` run. But that's basically the same.

Reply:
yeah exactly ... I actually rewrote it in the way you are proposing but then I actually realized it is more or less the same as fixed wait for that time and then assert...